### PR TITLE
[tensorflow] | [build] | [inference] Add dockerfiles for tf-2.3 inference

### PR DESCRIPTION
**Pull request by satishpasumarthi**
*Issue #, if available:*

## Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [build] | [test] | [build, test] | [ec2, ecs, eks, sagemaker]
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.
- [ ] I've updated the tensorflow-serving-*api versions to final instead of rc for final release.

#### EIA Checklist
* When creating a PR:
- [ ] I've modified ```src/config/build_config.py``` in my PR branch by setting ```ENABLE_EI_MODE = True```
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above

* For reviewer, before merging, please cross-check:
- [ ] I've verified the code change on the config file mentioned above has already been reverted

*Description:*
Docker files for TFS 2.3 [rc0 - prerelease]
*Tests run:*

*DLC image/dockerfile:*

*Additional context:*
FYI, newer versions of below packages are available:
cython - 0.29.21
gevent - 20.6.2
requests - 2.24.0
protobuf - 3.12.4
grpcio - 1.31.0 [updated already]
we have to decide if we want to go ahead with upgrading the above packages. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tensorflow/buildspec.yml [hunk 1]
```diff
@@ -44,6 +44,28 @@
       docker_file: !join [ docker/, *VERSION, /, *DOCKER_PYTHON_VERSION, /, *CUDA_VERSION, /Dockerfile., *DEVICE_TYPE ]
       context:
         <<: *TRAINING_CONTEXT
```
Comment (TusharKanekiDey) [inline tensorflow/buildspec.yml:46]: Can you add these lines in context section-
`    inference_context: &INFERENCE_CONTEXT
      sagemaker_package_name:
        source: docker/build_artifacts/sagemaker
        target: sagemaker
      deep_learning_container:
        source: docker/build_artifacts/deep_learning_container.py
        target: deep_learning_container.py
      init:
        source: docker/build_artifacts/__init__.py
        target: __init__.py
      dockerd-entrypoint:
        source: docker/build_artifacts/dockerd-entrypoint.py
        target: dockerd-entrypoint.py`

And these lines in repo_info-
`    inference_repository: &INFERENCE_REPOSITORY
      image_type: &INFERENCE_IMAGE_TYPE inference
      root: !join [ *FRAMEWORK, "/", *INFERENCE_IMAGE_TYPE ]
      repository_name: &REPOSITORY_NAME !join [pr, "-", *FRAMEWORK, "-", *INFERENCE_IMAGE_TYPE]
      repository: &REPOSITORY !join [ *ACCOUNT_ID, .dkr.ecr., *REGION, .amazonaws.com/, *REPOSITORY_NAME ]`

**Comment by TusharKanekiDey** [comment]
Thanks @satishpasumarthi for creating this PR. Will let you know soon if we should update the packages.

File: tensorflow/inference/docker/2.3.0/py3/Dockerfile.cpu [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+RUN apt-get update \
+ && apt-get -y install --no-install-recommends \
+    curl \
+    gnupg2 \
+    ca-certificates \
+    emacs \
+    git \
+    wget \
+    vim \
+    libbz2-dev \
```
Comment (arjkesh) [inline tensorflow/inference/docker/2.3.0/py3/Dockerfile.cpu:41]: please also add liblzma-dev -- see #467

File: tensorflow/inference/docker/2.3.0/py3/cu102/Dockerfile.gpu [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+# allow unauthenticated and allow downgrades for special libcublas library
+RUN apt-get update \
+ && apt-get install -y --no-install-recommends --allow-unauthenticated --allow-downgrades\
+    ca-certificates \
+    cuda-command-line-tools-10-2 \
+    cuda-cufft-10-2 \
+    cuda-curand-10-2 \
+    cuda-cusolver-10-2 \
+    cuda-cusparse-10-2 \
+    libbz2-dev \
```
Comment (arjkesh) [inline tensorflow/inference/docker/2.3.0/py3/cu102/Dockerfile.gpu:44]: please also add liblzma-dev -- see #467

File: tensorflow/buildspec.yml (hunk 1 above)
Comment (satishpasumarthi) [reply tensorflow/buildspec.yml:63]: Done.

File: tensorflow/buildspec.yml [hunk 4]
```diff
@@ -20,6 +25,18 @@
       deep_learning_container:
         source: docker/build_artifacts/deep_learning_container.py
         target: deep_learning_container.py
+      sagemaker_package_name:
```
Comment (YYStreet) [inline tensorflow/buildspec.yml:28]: Should add one line of `inference_context: &INFERENCE_CONTEXT`, between line 27 and line 28

Please refer https://github.com/aws/deep-learning-containers/blob/master/tensorflow/buildspec-tf1.yml#L28

Please take care of indentation.

Comment (satishpasumarthi) [reply tensorflow/buildspec.yml:32]: done.

**Pull request merged**